src/TesApi.Web/Controllers/TaskServiceApi.cs

```diff
@@ -287,28 +292,187 @@ public virtual async Task<IActionResult> GetTaskAsync([FromRoute][Required] stri
+                return BadRequest(exc.Message);
+            }
+
+            string nextPageToken = default;
+            IEnumerable<TesTask> tasks = default;
+
+            (nextPageToken, tasks) = await repository.GetItemsAsync(
+                decodedPageToken,
                 pageSize.HasValue ? (int)pageSize : 256,
-                decodedPageToken, cancellationToken);
+                cancellationToken,
+                rawPredicate,
+                predicate);
 
-            var encodedNextPageToken = nextPageToken is not null ? Base64UrlTextEncoder.Encode(Encoding.UTF8.GetBytes(nextPageToken)) : null;
+            var encodedNextPageToken = nextPageToken is null ? null : Base64UrlTextEncoder.Encode(Encoding.UTF8.GetBytes(nextPageToken));
             var response = new TesListTasksResponse { Tasks = tasks.ToList(), NextPageToken = encodedNextPageToken };
 
-            return TesJsonResult(response, view);
+            return TesJsonResult(response, viewEnum);
+        }
+
+        // ?tag_key=foo1&tag_value=bar1&tag_key=foo2&tag_value=bar2
+        private (IDictionary<string, string> Tags, string Error) ParseZipedTagsFromQuery()
+        {
+            // Per spec, every tag_value parameter must be preceded by exactly one corresponding tag_key parameter, and no tag_key parameter can be followed by more than one tag_value parameter.
+            // As a result, there can be a total number of tag_key parameters that are greater than the number of tag_value parameters. It's not legal for there to be more tag_value parameters.
+
+            var list = new List<(string Key, string Value)>();
+
+            foreach (var encodedPair in new QueryStringEnumerable(Request.QueryString.Value))
+            {
+                // Note: Microsoft's ASP.NET considers the names of items in the query string to be case insensitive, although the HTTP spec declares that they are normally considered to be case sensitive.
+                var name = new string(encodedPair.DecodeName().Span);
+
+                switch (name)
+                {
+                    case var x when "tag_key".Equals(x, StringComparison.OrdinalIgnoreCase):
+                        list.Add(new(new(encodedPair.DecodeValue().Span), null));
+                        break;
+
+                    case var x when "tag_value".Equals(x, StringComparison.OrdinalIgnoreCase):
+                        if (list.Count == 0 || list[^1].Value is not null)
+                        {
+                            return (null, "Unmatched tag_value");
+                        }
+
+                        list[^1] = (list[^1].Key, new(encodedPair.DecodeValue().Span));
+                        break;
+                }
+            }
+
+            // TODO: should tag keys be case sensitive or case insensitive? I would argue for case sensitive, as that is the default for JSON.
+            var tags = new Dictionary<string, string>(StringComparer.Ordinal); // StringComparer.OrdinalIgnoreCase
```

Comment:
is this the only place that needs to change for tag keys to be case insensitive?